src/infrastructure/db/sqlite/connection.rs

```diff
@@ -756,10 +756,20 @@ impl PlaceRepo for SqliteConnection {
         // that is required to filter the aggregated column.
         let mut sql = "SELECT tag, COUNT(*) as count \
                        FROM place_revision_tag \
-                       WHERE parent_rowid IN \
+                       WHERE {like_condition} parent_rowid IN \
```

Comment:
**!!!SQL INJECTION!!!** This is kind of code strongly discouraged and cannot be accepted.

Reply:
we are exposing the whole table anyways. btw, what do you suggest? i just need to receive the tags containing a particular string.

Reply:
Inserting user-defined strings literally and unescaped into an SQL query is not possible.

For individual columns you use value binding. For building a custom LIKE condition you need to properly escape the provided values to prevent SQL injections.

Reply:
@uklotzde now the characters get escaped. the changes are available on this PR and commit 10b2d99622d561b75d990376e7329a96825d16f3

Reply:
- What makes the crate you added as a dependency suitable for escaping SQL LIKE strings? It has been build for shell expressions and I don't think it is suitable the way it is used here.
- Tests are missing
- I am not convinced that we should do this at all. Even safely injecting parameters into SQL expressions, i.e. bypassing and ignoring all abstraction layers is not how services and APIs should be built.

@wellemut @flosse I was not aware of this new feature and only commented on the technical aspects. Since I don't know the actual requirements I cannot comment on how to implement it properly.

Reply:
- the necessity for this feature is to filter out tags containing a particular string to provide better suggestions.
- what crates do you suggest for this purpose?

Reply:
I am really sorry, not to be able to understand and comment this issue properly.
I just remember, that there was this "WASM-Modul" which should solve a best possible suggestion of tags for the frontend. The WASM Modul in the frontend should communicate better with the backend for highspeed suggestions.

Maybe the Wasm Modul needs to be better documented?

Reply:
with the option of filtering the tags in the backend there would be no need for extra modules in the front end. it means no extra overhead for installing wasm packages, storing any data on the client's device, and preventing more speed because the client will not need to match patterns.

Reply:
My suggestions would be to decide this in July when @flosse is back in a call. (together with the count of the bbox https://github.com/kartevonmorgen/openfairdb/issues/313)

Reply:
api/v0/entries/most-popular-tags?contains=test

Reply:
David implemented an own DB in the frontend-server, where he stores the tags.

But when @ziedavid is on the frontend, we will ask him to integrate the "WASM"-Module together with Markus.

If we then see a loss of speed, we still can integrate the  most-popular-tags?contains= in the backend-API